### PR TITLE
feat: api analytics grouped by cluster id

### PR DESCRIPTION
**Pull request by exu**
## Checklist (choose whats happened)

- [ ] breaking change! (describe)
- [ ] tested locally
- [ ] tested on cluster
- [ ] added new dependencies
- [ ] updated the docs
- [ ] added a test

## Breaking changes

-

## Changes

-

## Fixes

-

### CODE REVIEW

**Review by nicufk — APPROVED** [review]
👍

**Review by nicufk — COMMENTED** [review]
There are some linting warnings 😄

File: internal/pkg/api/repository/config/interface.go [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+	"context"
+
+	"github.com/kubeshop/testkube/pkg/api/v1/testkube"
+)
+
+type Repository interface {
+	// GetUniqueClusterId gets unique cluster based ID
+	GetUniuqueClusterId(ctx context.Context) (string, error)
+
+	// Get gets execution result by id
```
Comment (vsukhin) [inline internal/pkg/api/repository/config/interface.go:13]: typo

File: internal/pkg/api/repository/config/mongo.go [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+	}
+}
+
+type MongoRepository struct {
+	Coll *mongo.Collection
+}
+
+func (r *MongoRepository) GetUniqueClusterId(ctx context.Context) (clusterId string, err error) {
+	config := testkube.Config{}
+	_ = r.Coll.FindOne(ctx, bson.M{"id": Id}).Decode(&config)
```
Comment (vsukhin) [inline internal/pkg/api/repository/config/mongo.go:30]: I would check if it's other than not found err

**Review by vsukhin — APPROVED** [review]

**Pull request merged**